packages/x-charts/src/ChartsTooltip/ChartsTooltipContainer.tsx

```diff
@@ -73,15 +77,18 @@ function ChartsTooltipContainer(inProps: ChartsTooltipContainerProps) {
 
   const axisSystem = useAxisSystem();
 
-  const store = useStore<[UseChartCartesianAxisSignature]>();
-  const isOpen = useSelector(
+  const store = useStore<[UseChartCartesianAxisSignature, UseChartBrushSignature]>();
+  const hasData = useSelector(
     store,
     trigger === 'axis'
       ? (axisSystem === 'polar' && selectorChartsInteractionPolarAxisTooltip) ||
           (axisSystem === 'cartesian' && selectorChartsInteractionAxisTooltip) ||
           noAxis
       : selectorChartsInteractionItemIsDefined,
   );
+  const shouldPreventBecauseOfBrush = useSelector(store, selectorBrushShouldPreventTooltip);
+
+  const isOpen = !shouldPreventBecauseOfBrush && hasData;
```

Comment:
This will prevent the tooltip to do any useless selector computation

```suggestion
  const store = useStore<[UseChartCartesianAxisSignature, UseChartBrushSignature]>();
  const shouldPreventBecauseOfBrush = useSelector(store, selectorBrushShouldPreventTooltip);
  const hasData = useSelector(
    store,
    shouldPreventBecauseOfBrush ? () => null  : trigger === 'axis'
      ? (axisSystem === 'polar' && selectorChartsInteractionPolarAxisTooltip) ||
          (axisSystem === 'cartesian' && selectorChartsInteractionAxisTooltip) ||
          noAxis
      : selectorChartsInteractionItemIsDefined,
  );

  const isOpen = !shouldPreventBecauseOfBrush && hasData;
```

Reply:
Maybe it does not need to be a selector

```js

function getTooltipHasDataSelector(isPrevented, trigger, axisSystem){
	if (shouldPreventBecauseOfBrush) {
      return selectorReturnFalse;
    }
    if (trigger === 'item') {
      return selectorChartsInteractionItemIsDefined;
    }
    if (axisSystem === 'polar') {
      return selectorChartsInteractionPolarAxisTooltip;
    }
    if (axisSystem === 'cartesian') {
      return selectorChartsInteractionAxisTooltip;
    }
    return selectorReturnFalse;
}


const store = useStore<[UseChartCartesianAxisSignature, UseChartBrushSignature]>();
const shouldPreventBecauseOfBrush = useSelector(store, selectorBrushShouldPreventTooltip);
const hasData = useSelector(store, getTooltipHasDataSelector(shouldPreventBecauseOfBrush, trigger, axisSystem));
```